core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -176,6 +177,7 @@ private[spark] abstract class MemoryManager(
     if (conf.getBoolean("spark.memory.offHeap.enabled", false)) {
       require(conf.getSizeAsBytes("spark.memory.offHeap.size", 0) > 0,
         "spark.memory.offHeap.size must be > 0 when spark.memory.offHeap.enabled == true")
+      require(Platform.unaligned(), "No support for unaligned Unsafe")
```

Comment:
Could you add a better instruction for the user? E.g., tell them to set spark.memory.offHeap.enabled to false.